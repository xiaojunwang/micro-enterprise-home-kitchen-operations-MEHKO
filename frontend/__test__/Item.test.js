import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABC123',
  title: 'An item',
  price: 5555,
  description: 'A test item',
  image: 'testImage.jpg',
  largeImage: 'largeTestImage.jpg',
};

describe('<Item />', () => {
  it('Renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

// describe('<Item />', () => {
//   it('Renders image', () => {
//     const wrapper = shallow(<ItemComponent item={fakeItem} />);
//     const img = wrapper.find('img');
//     console.log(img.props());
//     expect(img.props().src).toBe(fakeItem.image);
//     expect(img.props().alt).toBe(fakeItem.title);
//   });
//   it('Renders price tag and title', () => {
//     const wrapper = shallow(<ItemComponent item={fakeItem} />);
//     const PriceTag = wrapper.find('PriceTag');
//     console.log(PriceTag.children().text());
//     expect(PriceTag.children().text()).toEqual('$99.99');
//     expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
//   });
//   it('Renders out buttons', () => {
//     const wrapper = shallow(<ItemComponent item={fakeItem} />);
//     console.log(wrapper.debug());
//     const buttonList = wrapper.find('.buttonList');
//     expect(buttonList.children()).toHaveLength(3);
//     expect(buttonList.find('Link')).toHaveLength(1);
//     expect(buttonList.find('AddToCart').exists()).toBe(true);
//     expect(buttonList.find('DeleteItem').exists()).toBe(true);
//   });
// });
