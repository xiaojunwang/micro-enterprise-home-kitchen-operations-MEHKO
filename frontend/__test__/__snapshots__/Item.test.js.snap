// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="An item"
    src="testImage.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        An item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $55.55
  </PriceTag>
  <p>
    A test item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
       Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
